### PR TITLE
[weather.ozweather] 0.9.8

### PR DESCRIPTION
**Pull request by bossanova808**
### Description

Updates for Leia and scraping changes

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [ x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] v1.0.0

### CODE REVIEW

**Comment by bossanova808** [comment]
Noted, but will update in the next update, thanks. Would like to get this out as it's a bit broken currently and not at my computer for a bit.

Thanks!

⁣Sent from TypeApp ​

**Comment by bossanova808** [comment]
@Rechi - not sure if you saw the above, forgot to @ you:

_Noted, but will update in the next update, thanks. Would like to get this out as it's a bit broken currently and not at my computer for a bit. Thanks!_

File: weather.ozweather/resources/lib/bom.py [hunk 1]
```diff
@@ -59,29 +59,36 @@ def downloadBackground(radarCode, fileName, backgroundsPath):
...
             imageFileRGB = backgroundsPath + outFileName
+            
             try:
+                #print(FTPSTUB + fileName)
                 image.retrieve(FTPSTUB + fileName, imageFileIndexed )
-            except:
+                # Needed due to bug in python 2.7 urllib - https://stackoverflow.com/questions/44733710/downloading-second-file-from-ftp-fails
+                urllib.urlcleanup()
+            except Exception as inst:
+                print(inst)
```
Comment (enen92) [inline weather.ozweather/resources/lib/bom.py]: please in the future drop all print statements and use only `xbmc.log`

**Comment by bossanova808** [comment]
Since I do unit testing using those prints, I presume it is ok if I leave one call to print in this context for that:

```
try:
    from xbmc import log as log
except ImportError:
    def log(str):
        print(str)
```

**Comment by bossanova808** [comment]
I've gone ahead and made the requested changes (leaving prints only in that function so I can still unit test) - and pushed these to my main repo (https://github.com/bossanova808/weather.ozweather).  I haven't changed the addon version number or anything.

I would like to update this PR with those new commits, but my git is fine for day to day stuff but this sub tree stuff seems to invariably go a bit wrong for me.

Would be great if you had some tips on how to take those changes here: https://github.com/bossanova808/weather.ozweather/commit/5708c6ff866450b0a7c265b577a36ebb87bcd7ac...and get them to appear here as updates to this PR.

Thanks for any help!

I could also nuke this PR and start from scratch with a new one (I've done that before when I've got into git trouble) - but would be good to know the right way....

**Comment by Rechi** [comment]
@bossanova808 yes I've seen your reply, but had no time to merge your update.

To update your PR execute the following commands (origin is your repo, upstream is the xbmc repo). The sub tree way (described in the wiki) is shit, IMO.

```
git fetch upstream
git checkout gotham
git reset --hard upstream/gotham
git rm -rf weather.ozweather/
# copy the addon source code into repo-scripts
git add -- weather.ozweather/
git commit -m '[weather.ozweather] 0.9.8'
git push -f origin gotham
```

**Comment by bossanova808** [comment]
@Rechi Awesome, thanks so much - and that approach is much simpler to follow for sure!  Will take a note of it for the future.

I've gone ahead and pushed all the requested changes.  Travis is happy at least!

**Comment by enen92** [comment]
Thanks for adjusting

**Pull request merged**